sycl/include/sycl/reduction.hpp

```diff
@@ -2317,44 +2329,26 @@ void reduction_parallel_for(handler &CGH,
                             nd_range<Dims> Range, PropertiesT Properties,
                             Reduction Redu, KernelType KernelFunc) {
   if constexpr (Reduction::has_float64_atomics) {
-    device D = detail::getDeviceFromHandler(CGH);
-
-    if (D.has(aspect::atomic64)) {
-      reduCGFuncForNDRangeBothFastReduceAndAtomics<KernelName>(
+    if (detail::getDeviceFromHandler(CGH).has(aspect::atomic64))
+      return reduCGFuncForNDRangeBothFastReduceAndAtomics<KernelName>(
           CGH, KernelFunc, Range, Properties, Redu);
-    } else {
-      reduction_parallel_for_basic_impl<KernelName>(
-          CGH, Queue, Range, Properties, Redu, KernelFunc);
-      return;
-    }
+
+    return reduction_parallel_for_basic_impl<KernelName>(
+        CGH, Queue, Range, Properties, Redu, KernelFunc);
   } else if constexpr (Reduction::has_fast_atomics) {
     if constexpr (Reduction::has_fast_reduce) {
-      reduCGFuncForNDRangeBothFastReduceAndAtomics<KernelName, KernelType>(
+      return reduCGFuncForNDRangeBothFastReduceAndAtomics<KernelName,
+                                                          KernelType>(
           CGH, KernelFunc, Range, Properties, Redu);
     } else {
-      reduCGFuncForNDRangeFastAtomicsOnly<KernelName, KernelType>(
+      return reduCGFuncForNDRangeFastAtomicsOnly<KernelName, KernelType>(
           CGH, KernelFunc, Range, Properties, Redu);
     }
   } else {
     reduction_parallel_for_basic_impl<KernelName>(CGH, Queue, Range, Properties,
                                                   Redu, KernelFunc);
     return;
```

Comment:
```suggestion
    return reduction_parallel_for_basic_impl<KernelName>(CGH, Queue, Range, Properties,
                                                  Redu, KernelFunc);
```
just to keep the pattern.